Core/Frameworks/Baikal/Core/Server.php

```diff
@@ -135,6 +135,8 @@ protected function initServer() {
             $authBackend = new \Baikal\Core\PDOBasicAuth($this->pdo, $this->authRealm);
         } elseif ($this->authType === 'Apache') {
             $authBackend = new \Sabre\DAV\Auth\Backend\Apache();
+        } elseif ($this->authType === 'LDAP') {
+            $authBackend = new \Baikal\Core\LDAP($this->pdo, 'users', $config['system']['ldap_mode'], $config['system']['ldap_uri'], $config['system']['ldap_bind_dn'], $config['system']['ldap_bind_password'], $config['system']['ldap_dn'], $config['system']['ldap_cn'], $config['system']['ldap_mail'], $config['system']['ldap_search_base'], $config['system']['ldap_search_attribute'], $config['system']['ldap_search_filter'], $config['system']['ldap_group']);
```

Comment:
With an LDAP-specific settings section, you could pass that full section in a single parameter instead of extracting this huge number of settings here

Reply:
I don't quite get what you mean, that parameter would still need to read the values from the config file, resulting in even larger code.

Reply:
I was thinking about something like `$authBackend = new \Baikal\Core\LDAP($this->pdo, 'users', $config['ldap]);`. The constructor internally needs a ton of lines to assign everything to member variables anyway.

Alternatively, how about using an `LdapConfig` struct? Something like this:

```
class LdapConfig {
    public $ldap_mode;
    public $ldap_uri;
    ...
}
```

```
$LdapConfig = new LdapConfig();
$LdapConfig->ldap_uri = $config['xy];
...
$authBackend = new \Baikal\Core\LDAP($this->pdo, 'users', $LdapConfig);
```

```
class LDAP extends \Sabre\DAV\Auth\Backend\AbstractBasic {
    protected $LdapConfig;
    public function __construct(..., $LdapConfig) {
        $this->LdapConfig = $LdapConfig;
    }
}
```

Reply:
@ByteHamster, I see what you mean, but as LDAP config is set in "System Settings", it'll be saved under `$config['system']`. So either we create a new tab or we could add code in `set` to handle LDAP set the values independently.

Reply:
Yeah, an additional settings section sounds like a lot of work and also quite hacky. How do you feel about that "struct"? It would reduce the long list of constructor parameters, making it a bit easier to comprehend

Reply:
@ByteHamster, is this ok?